packages/block-library/src/heading/index.js

```diff
@@ -30,15 +30,16 @@ export const settings = {
 		const { content, level } = attributes;
 
 		const customName = attributes?.metadata?.name;
+		const hasContent = content?.length > 0;
```

Comment:
This will work for both `undefined` and `RichTextData`.

Reply:
Nice, that's handy! TIL the `RichTextData` class has [a `length` getter](https://github.com/WordPress/gutenberg/blob/d1bbd9f9badb5a24866621ec9b4b0afb83bbb068/packages/rich-text/src/create.js#L162) 👍